### PR TITLE
Setup redis and add caching

### PR DESCRIPTION
**Pull request by MaximeNdutiye**
Add Redis so we can cache records.

For testing locally:
`brew install redis`

Start Redis
`brew services start redis`

Create some perks using the front end ui

I've made some changes the perks controller to log whenever there is a cache miss.
Now if we visit /perks/1 we can see that on the first call there will be a cache miss. 
On subsequent calls the log will not appear meaning that that we hit the cache.

image

### CODE REVIEW

**Comment by zachary-nguyen** [comment]
@MaximeNdutiye I think with memcached and Redis using the same Spring annotations lets deploy two separate servers. One containing Memcached and one Redis. This will ensure we don't have any issues with two datastores.

File: .travis.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ env:
   - NODE_VERSION="14.16.0"
 services:
   - postgresql
+  - redis-server
```
Comment (zachary-nguyen) [inline .travis.yml:9]: Is this all we need to enable redis in heroku?

Comment (MaximeNdutiye) [reply .travis.yml:9]: That just adds Redis to our Travis CI builds. I provisioned our heroku redis instance using 
`heroku addons:create heroku-redis:hobby-dev -a perk-manager` as described here: https://devcenter.heroku.com/articles/heroku-redis

**Comment by zachary-nguyen** [comment]
LGTM 👍

**Review by zachary-nguyen — APPROVED** [review]

**Review by fareed-ah — APPROVED** [review]

**Pull request merged**